### PR TITLE
fix(linux): work around Lintian errors

### PR DESCRIPTION
**Pull request by ermshiperete**
We only use a C API in Core, so we can safely ignore all C++ symbols that show up in the symbols file. Those come from std template instantiations.

Also update AppStream metadata.

Fixes: #12051 

@keymanapp-test-bot skip

### CODE REVIEW

File: linux/debian/com.keyman.config.appdata.xml [hunk 1]
```diff
@@ -5,10 +5,15 @@
   <metadata_license>MIT</metadata_license>
   <project_license>MIT</project_license>
   <project_group>Keyman</project_group>
-  <developer_name>SIL International</developer_name>
+  <developer id="org.sil.keyman">
+    <name>SIL International</name>
```
Comment (mcdurdin) [inline linux/debian/com.keyman.config.appdata.xml:9]: SIL Global?

File: linux/debian/com.keyman.ibus_keyman.metainfo.xml [hunk 2]
```diff
@@ -2,9 +2,11 @@
 <component type="inputmethod">
   <id>com.keyman.ibus_keyman</id>
   <metadata_license>MIT</metadata_license>
-  <project_license>GPL-2+</project_license>
+  <project_license>GPL-2.0+</project_license>
   <project_group>Keyman</project_group>
-  <developer_name>SIL International</developer_name>
+  <developer id="org.sil.keyman">
+    <name>SIL International</name>
```
Comment (mcdurdin) [inline linux/debian/com.keyman.ibus_keyman.metainfo.xml:8]: SIL Global?

File: linux/debian/com.keyman.config.appdata.xml (hunk 1 above)
Comment (ermshiperete) [reply linux/debian/com.keyman.config.appdata.xml]: Ah, right. Done.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Pull request merged**